code/game/objects/items/weapons/traps.dm

```diff
@@ -123,10 +124,519 @@
+/obj/item/weapon/trap/animal/proc/mob_breakout(var/mob/living/escapee)
+	if (req_breakout() < 1)
+		return
+
+	escapee.next_move = world.time + 100
+	escapee.last_special = world.time + 100
+	to_chat(escapee, "<span class='warning'>You to shake and bump the lock of \the [src]. (this will take about [time_to_escape] minutes).</span>")
+	visible_message("<span class='danger'>\The [src] begins to shake violently! Something is attempting to escape it!</span>")
+
+	var/time = 360 * time_to_escape * 2
+	breakout = TRUE
+
+	if (!do_after(escapee, time, act_target = src, extra_checks = CALLBACK(src, .proc/breakout_callback, escapee)))
+		breakout = FALSE
+		return
+
+	breakout = FALSE
+	to_chat(escapee, "<span class='warning'>You successfully break out!</span>")
+	visible_message("<span class='danger'>\the [escapee] successfully broke out of \the [src]!</span>")
+	playsound(loc, "sound/effects/grillehit.ogg", 100, 1)
+	if(!contents.len)
+		return
+
+	for(var/mob/living/L in contents)
+		L.forceMove(src.loc)
+		visible_message("<span class='warning'>[L] runs out of \the [src].</span>")
+		deployed = 0
+		update_icon()
+		src.animate_shake()
+		release_time = world.time
+
+	for(var/obj/effect/spider/spiderling/S in contents) // for spiderlings
+		S.forceMove(src.loc)
+		START_PROCESSING(SSprocessing, S)
+		visible_message("<span class='warning'>[S] jumps out of \the [src].</span>")
+		deployed = 0
+		update_icon()
+		src.animate_shake()
+		release_time = world.time
+
+/obj/item/weapon/trap/animal/Collide(AM as mob|obj)
+	if(isliving(AM))
+		Crossed(AM)
+	else
+		..()
+
+/obj/item/weapon/trap/animal/CollidedWith(atom/AM)
+	if(isliving(AM))
+		Crossed(AM)
+	else
+		..()
+
+/obj/item/weapon/trap/animal/verb/release_animal()
+	set src in oview(1)
+	set category = "Object"
+	set name = "Release animal"
+
+	if(!usr.canmove || usr.stat || usr.restrained())
+		return
+
+	if(!ishuman(usr))
+		to_chat(usr, "<span class='warning'>This mob type can't use this verb.</span>")
+		return
+
+	if(deployed)
+		var/open = alert("Do you want to open the cage and free what is inside?",,"No","Yes")
+
+		if(open == "No")
+			return
+
+		if(!can_use(usr))
+			to_chat(usr, "<span class='warning'>You cannot use \the [src].</span>")
+			return
+		if(!contents.len)
+			return
+
+		var/turf/T_cage = get_turf(src)
+		var/turf/T_user= get_turf(usr)
+
+		if(!T_cage)
+			attack_self(src)
+			return
+
+		var/turf/target = get_turf(locate(T_cage.x + (T_cage.x - T_user.x), T_cage.y + (T_cage.y - T_user.y), T_cage.z))
+		if(!target)
+			attack_self(src)
+			return
+
+		visible_message("<span class='notice'>[usr] opens \the [src].</span>")
+		for(var/mob/living/L in contents)
+			L.forceMove(target)
+			visible_message("<span class='warning'>[L] runs out of \the [src].</span>")
+			animate_shake()
+			deployed = 0
+			update_icon()
+			src.animate_shake()
+			release_time = world.time
+
+		for(var/obj/effect/spider/spiderling/S in contents) // for spiderlings
+			S.forceMove(target)
```

Comment:
Code Duplication - Release
Extract all of them into a function with a parameter for the visible message and a optional parameter to start processing